### PR TITLE
#4802 - Not render checkbox for free payment

### PR DESCRIPTION
**Pull request by IrinaZhadzinets**
**Includes:**
* https://github.com/scandipwa/quote-graphql/pull/100

**Related issue(s):**
* Fixes https://github.com/scandipwa/scandipwa/issues/4802

**Problem:**
* All enabled payment methods are shown on billing step if price total 0

**In this PR:**
* Only 'Zero Subtotal Checkout' payment method is shown on billing step if price total 0

### CODE REVIEW

File: packages/scandipwa/src/component/CheckoutPayments/CheckoutPayments.component.js [hunk 1]
```diff
@@ -113,7 +113,19 @@ export class CheckoutPayments extends PureComponent {
...
+        if (paymentMethods.length === 1 && paymentMethods[0].code === 'free') {
+            const { code, title } = paymentMethods[0];
+
+            selectPaymentMethod({ code });
+
+            return (
+                <li block="CheckoutPayment">
+                    { title }
+                </li>
+            );
```
Comment (zans-laksa) [inline packages/scandipwa/src/component/CheckoutPayments/CheckoutPayments.component.js:127]: Please call `renderPayment` with the single payment method item instead.

Comment (zans-laksa) [reply packages/scandipwa/src/component/CheckoutPayments/CheckoutPayments.component.js:127]: Actually you can allow `return paymentMethods.map(this.renderPayment);` run.
As I understand, you just have to preselect the payment method here?

Comment (IrinaZhadzinets) [reply packages/scandipwa/src/component/CheckoutPayments/CheckoutPayments.component.js:127]: The main difference is that paymentMethods draws items with a checkbox, and if use it, need to do the same check in paymentMethods, and still display just text

Is it better to do this check and other rendering in paymentMethods?

Comment (zans-laksa) [reply packages/scandipwa/src/component/CheckoutPayments/CheckoutPayments.component.js:127]: I meant that even if we filter out other payment methods and have only one, it should have the same appearance as they would be more than one.
But I'm not a designer and may be wrong. @carinadues please correct if I'm wrong.

Comment (IrinaZhadzinets) [reply packages/scandipwa/src/component/CheckoutPayments/CheckoutPayments.component.js:127]: Just in the description of the PR that Marita did, it is indicated that the checkbox needs to be removed(https://github.com/scandipwa/scandipwa/pull/4817#issuecomment-1144350185)

**Review by zans-laksa — APPROVED** [review]

**Comment by aleksandrakorolova** [comment]
All payment methods are displayed on billing step if user is buying NOT virtual product and selected free shiping
Screenshot 2022-07-28 at 16 47 58

**Notes:**
After refreshing page - works as expected

**Comment by IrinaZhadzinets** [comment]
Fix in https://github.com/scandipwa/quote-graphql/pull/102

**Comment by aleksandrakorolova** [comment]
Can't check PR because product isn't added to cart
message: "Cannot query field \"getCartForCustomer\" on type \"Query\"."
Screenshot 2022-08-02 at 09 16 37 (2)

**Pull request merged**